python/ray/services.py

```diff
@@ -808,6 +822,21 @@ def start_ray_processes(address_info=None,
                                       stderr_file=redis_stderr_file,
                                       cleanup=cleanup)
       assert redis_port == new_redis_port
+
+    redis_client = redis.StrictRedis(host=node_ip_address, port=redis_port)
+    # Start other Redis shards listening on random ports.
+    for i in range(num_redis_shards):
+      redis_port, _ = start_redis(node_ip_address,
+                                  stdout_file=redis_stdout_file,
+                                  stderr_file=redis_stderr_file,
+                                  cleanup=cleanup)
+      redis_shards.append(address(node_ip_address, redis_port))
+    address_info["redis_shards"] = redis_shards
+    # Store redis shard information in the primary redis shard.
+    for shard in redis_shards:
+      redis_client.rpush("RedisShards", shard)
```

Comment:
Since we're storing redis shard information in the primary redis shard anyway, maybe we can avoid passing around the shard information to all the components, and have them look up the shard addresses on startup?